cmd/containerd-shim-runc-v2/process/init.go

```diff
@@ -209,8 +209,11 @@ func (p *Init) createCheckpointedState(r *CreateConfig, pidFile *pidFile) error
 }
 
 // Wait for the process to exit
-func (p *Init) Wait() {
-	<-p.waitBlock
+func (p *Init) Wait(ctx context.Context) {
```

Comment:
It looks good. would you please add one e2e test, like when that client cancel that Wait, the task should be still running and stoppable. Thanks